### ISSUE TITLE
App not starting on Ubuntu 22.04

### ISSUE THREAD

**Issue by Maltergate** [mjakeman/extension-manager#237]
Ubuntu 22.04 / Gnome 42.2
App installed with APT package manager
```
>> apt-cache policy gnome-shell-extension-manager
gnome-shell-extension-manager:
  Installed: 0.3.0-0ubuntu2.1
  Candidate: 0.3.0-0ubuntu2.1
  Version table:
 *** 0.3.0-0ubuntu2.1 500
        500 http://nz.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages
        100 /var/lib/dpkg/status
     0.3.0-0ubuntu1 500
        500 http://nz.archive.ubuntu.com/ubuntu jammy/universe amd64 Packages
```

The App doesn't launch. When launching from a terminal, I get the following:
```
>> extension-manager 

(extension-manager:5052): GLib-CRITICAL **: 10:11:09.904: the GVariant format string 'd' has a type of 'd' but the given value has a type of 's'

(extension-manager:5052): GLib-CRITICAL **: 10:11:09.905: g_variant_get: assertion 'valid_format_string (format_string, TRUE, value)' failed

(extension-manager:5052): Gtk-WARNING **: 10:11:09.928: Failed to set text 'Sound Input & Output Device Chooser' from markup due to error parsing markup: Error on line 1: Entity did not end with a semicolon; most likely you used an ampersand character without intending to start an entity — escape ampersand as &amp;
Gdk-Message: 10:11:10.236: Error flushing display: Protocol error
```

Bug also reported on the Ubuntu Launchpad: https://bugs.launchpad.net/bugs/1984143

**Comment by mjakeman** [comment]
That's bizarre... Does uninstalling and reinstalling the app help, or is it always broken?

Are you doing anything 'interesting' on your system, like running Nvidia drivers?

**Comment by Maltergate** [comment]
Bingo! Nice flair.

I do have nvidia-drivers on my system (`nvidia-smi` gives me ver. 515). I am running Ubuntu 22.04 on Wayland. When logging with Xorg, the app starts without issue. Launching in command-line still gives the critical Glib errors:

```
>>extension-manager 

(extension-manager:45301): GLib-CRITICAL **: 18:42:09.293: the GVariant format string 'd' has a type of 'd' but the given value has a type of 's'

(extension-manager:45301): GLib-CRITICAL **: 18:42:09.293: g_variant_get: assertion 'valid_format_string (format_string, TRUE, value)' failed

(extension-manager:45301): Gtk-WARNING **: 18:42:09.302: Failed to set text 'Sound Input & Output Device Chooser' from markup due to error parsing markup: Error on line 1: Entity did not end with a semicolon; most likely you used an ampersand character without intending to start an entity — escape ampersand as &amp;
```

So, Wayland seems to be the bothering bit here.

**Comment by mjakeman** [comment]
> I do have nvidia-drivers on my system (nvidia-smi gives me ver. 515). I am running Ubuntu 22.04 on Wayland.

Ah that's good to hear. Ironically this actually makes it easier to come to a solution since it's a fairly common issue from what I can tell. For now, let's merge this with #184 which is a catch-all for Nvidia-related problems.

Regarding the Glib errors, those aren't anything to worry about - they should be fixed in 0.3.1.
